pyflakes/checker.py

```diff
@@ -507,6 +513,20 @@ def __init__(self, tree, filename='(none)', builtins=None,
         self.popScope()
         self.checkDeadScopes()
 
+    @staticmethod
+    def getScope(tree):
+        if isinstance(tree, ast.Module):
```

Comment:
define a _private class mapping
```py
_ast_node_scope = {
    ast.Module: ModuleScope,
    ast.ClassDef: ClassScope,
    ast.FunctionDef: FunctionScope,
    ast.Lambda: FunctionScope,
    ...
}
```

Then this lookup is easy to do (and doesnt require new function), and `_ast_node_scope.keys()` can be used in `getScopeNode`.